### PR TITLE
[JG] OPS-12826 - adding new parts to charges object, split object into two

### PR DESCRIPTION
**Pull request by DreamcastKnight**

### CODE REVIEW

File: cor-journey/src/main/scala/essttp/rootmodel/ttp/eligibility/Charges.scala [hunk 1]
```diff
@@ -16,31 +16,54 @@
...
-    outstandingAmount:             OutstandingAmount,
-    interestStartDate:             Option[InterestStartDate],
-    dueDate:                       DueDate,
-    accruedInterest:               AccruedInterest,
-    ineligibleChargeType:          IneligibleChargeType,
-    chargeOverMaxDebtAge:          Option[ChargeOverMaxDebtAge],
-    locks:                         Option[List[Lock]],
-    dueDateNotReached:             Boolean,
-    isInterestBearingCharge:       Option[IsInterestBearingCharge],
+final case class Charges(charges1: Charges1, charges2: Charges2)
```
Comment (AndyHWChung) [inline cor-journey/src/main/scala/essttp/rootmodel/ttp/eligibility/Charges.scala:23]: don't we just need to keep the original case class and add the optional fields to it?

Comment (AndyHWChung) [reply cor-journey/src/main/scala/essttp/rootmodel/ttp/eligibility/Charges.scala:23]: oh, just saw your message about the size limit - what was the error with the size limit you were seeing?

Comment (DreamcastKnight) [reply cor-journey/src/main/scala/essttp/rootmodel/ttp/eligibility/Charges.scala:23]: there's a 22 object limit, so the class needs to be split up or it breaks

File: cor-journey/src/main/scala/essttp/rootmodel/ttp/eligibility/Charges.scala [hunk 2]
```diff
@@ -58,7 +58,7 @@ final case class Charges2(
 object Charges {
   @SuppressWarnings(Array("org.wartremover.warts.Any"))
   implicit val writes: Writes[Charges] = Writes { charge =>
-    Json.obj(
+    Json.obj("Charges" -> Json.obj(
```
Comment (AndyHWChung) [inline cor-journey/src/main/scala/essttp/rootmodel/ttp/eligibility/Charges.scala:61]: do you need the outer `"Charges"` bit? Couldn't it just be
```
Json.obj(
   "chargeType" -> ...
...
)
```
That's how it'd appear from TTP anyways I think

**Review by AndyHWChung — APPROVED** [review]

**Review by AndyHWChung — APPROVED** [review]

**Pull request merged**